public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -301,16 +304,21 @@ export function lokiStreamsToDataframes(
   data: LokiStreamResult[],
   target: { refId: string; expr?: string; regexp?: string },
   limit: number,
+  config: LokiOptions,
   reverse = false
 ): DataFrame[] {
-  const series: DataFrame[] = data.map(stream => ({
-    ...lokiStreamResultToDataFrame(stream, reverse),
-    refId: target.refId,
-    meta: {
-      searchWords: getHighlighterExpressionsFromQuery(formatQuery(target.expr, target.regexp)),
-      limit,
-    },
-  }));
+  const series: DataFrame[] = data.map(stream => {
+    const dataFrame = lokiStreamResultToDataFrame(stream, reverse);
+    enhanceDataFrame(dataFrame, config);
```

Comment:
This does not seem the work, though the `dataFrame ` is enhanced with the derived fields it is not actually used afterwards as `lokiStreamResultToDataFrame` creates a new dataFrame.

Reply:
my mistake, forgot to replace `lokiStreamResultToDataFrame` with `dataFrame` in the next line 😛